kolena/workflow/evaluator.py

```diff
@@ -268,22 +270,25 @@ def _validate_metrics_test_sample_type(metrics_test_sample_type: Type[MetricsTes
 
 
 def _validate_metrics_test_case_type(metrics_test_case_type: Type[DataObject]) -> None:
-    validate_scalar_data_object_type(metrics_test_case_type, supported_list_types=[MetricsTestCase])
+    validate_scalar_data_object_type(
+        metrics_test_case_type,
+        supported_list_types=[MetricsTestCase, ThresholdedMetrics],
+    )
 
     # validate that there is only one level of nesting
     for field_name, field_type in get_data_object_field_types(metrics_test_case_type).items():
         origin = get_origin(field_type)
-        if origin is not list:  # only need to check lists, as MetricsTestCase is only allowed in lists
-            continue
+
         # expand e.g. List[Union[MetricsA, MetricsB]] into [MetricsA, MetricsB]
-        list_arg_types = [t for arg_type in get_args(field_type) for t in get_args(arg_type) or [arg_type]]
-        for arg_type in list_arg_types:
-            if arg_type is None:
-                raise ValueError(f"Unsupported optional metrics object in field '{field_name}'")
-            try:
-                validate_scalar_data_object_type(arg_type)
-            except ValueError:
-                raise ValueError(f"Unsupported doubly-nested metrics object in field '{field_name}'")
+        if origin is list:
```

Comment:
The logic here doesn't seem to have changed:
```
# old
if origin is not list:  # only need to check lists, as MetricsTestCase is only allowed in lists
    continue

# current
if origin is list:
    ...
```

Reply:
ok rollback to the original code which is equivalent to this; agree not necesary to rewrite if we are not adding any change.